src/ark/segmentation/fiber_segmentation.py

```diff
@@ -294,9 +354,194 @@ def segment_fibers(data_xr, fiber_channel, out_dir, fov, blur=2, contrast_scalin
+                represents the fiber mask
+            fov_length (int):
+                length of the image
+            tile_length (int):
+                length of tile size, must be a factor of the total image size (default 512)
+            min_fiber_num (int):
+                the amount of fibers to get tile statistics calculated, if not then NaN (default 5)
+            save_dir (str):
+                directory where to save tiled image folder to
+            save_tiles (bool):
+                whether to save cropped images (default to False)
+
+        Returns:
+            pd.DataFrame:
+             - a dataframe specifying each tile in the image and its calculated stats
+        """
+
+    fov_table = fov_table.reset_index(drop=True)
+    fov = fov_table.fov[0]
+    alignment, length, pixel_density, fiber_density = [], [], [], []
+    fov_list, tile_x, tile_y = [], [], []
+
+    # create tiles based on provided tile_length
+    for i, j in itertools.product(
+            range(int(fov_length / tile_length)), range(int(fov_length / tile_length))):
+        y_range = (i * tile_length, (i + 1) * tile_length)
+        x_range = (j * tile_length, (j + 1) * tile_length)
+
+        fov_list.append(fov)
+        tile_x.append(x_range[0])
+        tile_y.append(y_range[0])
+
+        if save_tiles:
+            tile_fiber_img = fov_fiber_img[y_range[0]:y_range[1], x_range[0]:x_range[1]]
+            tile_fiber_img[tile_fiber_img > 0] = 1
+            if not os.path.exists(os.path.join(save_dir, fov)):
+                os.makedirs(os.path.join(save_dir, fov))
+            io.imsave(os.path.join(save_dir, fov, f'tile_{y_range[0]},{x_range[0]}.tiff'),
+                      tile_fiber_img, check_contrast=False)
+
+        # subset table for only fibers within the tile coords
+        tile_table = fov_table[np.logical_and(
+            fov_table['centroid-0'] >= y_range[0], fov_table['centroid-0'] < y_range[1])]
+        tile_table = tile_table[np.logical_and(
+            tile_table['centroid-1'] >= x_range[0], tile_table['centroid-1'] < x_range[1])]
+
+        # tile must have a certain number of fibers to receive values, otherwise NaN
+        if len(tile_table) < min_fiber_num:
+            for stat_list in [alignment, length, pixel_density, fiber_density]:
+                stat_list.append(np.nan)
+        else:
+            # alignment
+            align_scores = tile_table['alignment_score'].values
+            align_scores = align_scores[~np.isnan(align_scores)]
+            avg_alignment = np.mean(align_scores) if len(align_scores) >= min_fiber_num else np.nan
+            alignment.append(avg_alignment)
+
+            # length
+            avg_length = np.mean(tile_table['major_axis_length'].values)
+            length.append(avg_length)
+
+            # density
+            p_density, f_density = calculate_density(tile_table, tile_length ** 2)
+            pixel_density.append(p_density)
+            fiber_density.append(f_density)
```

Comment:
You could condense this `if/else` statement a bit:

```
avg_alignment, avg_length, p_density, f_density = np.nan

if len(tile_table) >= min_fiber_num:
    align_scores = tile_table['alignment_score'].values
    align_scores = align_scores[~np.isnan(align_scores)]
    avg_alignment = np.mean(align_scores) if len(align_scores) >= min_fiber_num else np.nan

    avg_length = np.mean(tile_table['major_axis_length'].values)

    p_density, f_density = calculate_density(tile_table, tile_length ** 2)

alignment.append(avg_alignment)
length.append(avg_length)
pixel_density.append(p_density)
fiber_density.append(f_density)
```

This personally makes the workflow a bit more clear for me, but it's up to you what works best.

Reply:
Totally agree, this is more straight forward!